### ISSUE TITLE
createWebhook and webhook.edit's avatar returns null

### ISSUE THREAD

**Issue by SmokyBird** [discordjs/discord.js#4368]
**Please describe the problem you are having in as much detail as possible:** When I try to create a webhook (or edit it) and sending an url of the image, both promises returns `avatar: null`. The webhook is created, but there's no avatar attached to it.

**Include a reproducible code sample here, if possible:**

```js
let webhookAvatar = Array.from(message.attachments)[0][1].proxyURL;

guild.channels
  .get(webhookChannelID)
  .createWebhook("test-1", webhookAvatar)

  .then((webhookCreated) => {
    webhookCreated
      .edit({ name: "test-2", avatar: webhookAvatar })

      .then((webhookEdited) => {
        console.log(webhookEdited);
      })

      .catch((error) => {
        console.log(error);
      });
  })

  .catch((error) => {
    console.log(error);
  });
```

**Further details:**

- discord.js version: 11.6.4
- Node.js version: 12.16.1
- Operating system: Windows 10
- Priority this issue should have – please be realistic and elaborate if possible: Minor

**Comment by ChiefChippy2** [comment]
Please correct the syntax errors in your code

**Comment by SmokyBird** [comment]
What do you mean, by syntax errors?

**Comment by aabbccsmith** [comment]
@SmokyBird I believe they are referencing the poor code style. Maybe a tool like [prettier](https://prettier.io/) might help? You can use it online [here](https://prettier.io/playground/).

**Comment by SmokyBird** [comment]
Not sure what it changes, since I've always worked this way, but yeah, okay, changed it anyway.

**Comment by Monbrey** [comment]
This is all pointless and off topic discussion.

The issue tracker is only for bug reports and enhancement suggestions. If you have a question, please ask it in the [Discord server](https://discord.gg/bRCvFy9) instead of opening an issue – you will get redirected there anyway.

**Comment by SmokyBird** [comment]
@Monbrey I've actually asked twice on Discord, and got no answer for it, but thanks for the suggestion.
I've created the issue because using `.createWebhook` and `webhook.edit` only creates/edits the webhook with the name specified `"test-1"`, but passing the `avatar` parameter doesn't work at all.

I've tried with the first image link attached to the message, direct image link from internet (even with examples specified in the docs) also base64 ; nothing work, webhook's promise tells me that `avatar` is null, and the webhook stays with the default avatar.

**Comment by Monbrey** [comment]
I'm unable to replicate this. Using an Imgur image URL I was able to edit the webhooks avatar.

Can you provide the actual URL you're using to edit it with? I'm thinking maybe it's format related or similar.

**Comment by SmokyBird** [comment]
I've tried with:
- [The link provided in discord.js' docs](https://i.imgur.com/mI8XcpG.jpg)
- [The proxyURL of the image attached to the message](https://cdn.discordapp.com/attachments/648701429305049126/723470195535904799/paranoia-test.png)

**Comment by Monbrey** [comment]
Still unable to replicate - edited a webhook to the first image, and sent a message.
![image](https://user-images.githubusercontent.com/5294381/85200103-f00a4700-b337-11ea-9a4b-cfe62ce276f3.png)
`client.fetchWebhook("723845438746394624").then(w => w.edit({ avatar: "https://i.imgur.com/mI8XcpG.jpg" }).then(w2 => w2.send("Test message")));`

Either way, any bugs in v11 aren't going to be fixed, as that version isn't maintained anymore. Please upgrade to v12 and see if the issue persists.

**Comment by SmokyBird** [comment]
I've tried those workaroundsto see if it could worked.
1)
- Create a webhook and return his ID
- Fetch the webhook
- Edit the webhook with the option `{ avatar : "https://i.imgur.com/mI8XcpG.jpg" }`
- Send a message with the webhook fetched

2)
- Fetch a webhook already created in the guild
- Edit the webhook with the option `{ avatar : "https://i.imgur.com/mI8XcpG.jpg" }`
- Send a message with the webhook fetched

Both workarounds didn't work, so I guess the problem might come from the version itself.

I'm not sure if someone actually reported the problem before, and checking v12's changelog, I didn't see any kind of fix for it, if there was ; and upgrading to v12 might take me a lot of time (but that's on my own, of course).

**Comment by SmokyBird** [comment]
Welp. For any reasons unknown to me, it works now, even if the code didn't change at all. There wasn't any update at all of discord.js' version or Discord's API.

I'm not really sure why it does that, the only *difference* I can see and tell is that I use different ways to developing a bot, and host it online.
For the developing part, the bot is hosted with my browser (Firefox 77.0.1) using the [11.6.4.min web build](https://github.com/discordjs/discord.js/tree/webpack). And this is exactly why I've created the issue: it didn't work as expected in this build I'm using actually, while it actually worked when it is hosted online.

Sorry for the inconvenience, and also thanks for the help provided.
I hope that someone who's in the same struggle as me will notice this and won't fall in that trap as I did. I usually wait that all my code works before hosting it online, and I guess my error was there: not testing further and better check where the problem comes from.
